File: OOП Lab 1.2/Lab1.2/Segment.cpp

#define _USE_MATH_DEFINES
#include "Segment.h"
#include <cmath>

Segment::Segment() : x1(0), y1(0), x2(0), y2(0) {}

Segment::Segment(double x1, double y1, double x2, double y2)
    : x1(x1), y1(y1), x2(x2), y2(y2) {}

Segment::Segment(const Segment& other)
    : x1(other.x1), y1(other.y1), x2(other.x2), y2(other.y2) {}

Segment::~Segment() {}

double Segment::length() const {
    return std::sqrt(std::pow(x2 - x1, 2) + std::pow(y2 - y1, 2));
}

double Segment::angleWithOY() const {
    double deltaY = y2 - y1;
    double deltaX = x2 - x1;
    return std::atan2(deltaX, deltaY) * (180.0 / M_PI);
}

void Segment::printInfo() const {
    std::cout << "Segment starts at (" << x1 << ", " << y1 << ") and ends at (" << x2 << ", " << y2 << ").\n";
    std::cout << "Length: " << length() << "\n";
    std::cout << "Angle with OY: " << angleWithOY() << " degrees\n";
}

double Segment::getX1() const { return x1; }
double Segment::getY1() const { return y1; }
double Segment::getX2() const { return x2; }
double Segment::getY2() const { return y2; }

File: OOП Lab 1.2/Lab 1.2/Segment.cs
using System;

namespace SegmentTask
{
    public class Segment
    {
        private double x1, y1, x2, y2;

        public Segment()
        {
            x1 = y1 = x2 = y2 = 0;
        }

        public Segment(double x1, double y1, double x2, double y2)
        {
            this.x1 = x1;
            this.y1 = y1;
            this.x2 = x2;
            this.y2 = y2;
        }

        public Segment(Segment other)
        {
            x1 = other.x1;
            y1 = other.y1;
            x2 = other.x2;
            y2 = other.y2;
        }

        ~Segment()
        {
            Console.WriteLine("Segment object is being deleted");
        }

        public double Length()
        {
            return Math.Sqrt(Math.Pow(x2 - x1, 2) + Math.Pow(y2 - y1, 2));
        }

        public double AngleWithOY()
        {
            double deltaY = y2 - y1;
            double deltaX = x2 - x1;
            return Math.Atan2(deltaX, deltaY) * (180 / Math.PI);
        }

        public void PrintInfo()
        {
            Console.WriteLine($"Segment starts at ({x1}, {y1}) and ends at ({x2}, {y2}).");
            Console.WriteLine($"Length: {Length():F2}");
            Console.WriteLine($"Angle with OY: {AngleWithOY():F2} degrees");
        }

        public double GetX1() => x1;
        public double GetY1() => y1;
        public double GetX2() => x2;
        public double GetY2() => y2;
    }
}

